app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt

```diff
@@ -55,8 +55,9 @@ class DroneExecutorImpl(
         val disarmed = changeFromTo(instance.telemetry.armed)
                 .doOnComplete { throw Error(ctx.getString(R.string.drone_disarmed_during_setup)) }
 
+        data.getMutableDroneStatus().postValue(IDLE)
+
```

Comment:
Why does the drone recieve the mission after having taking off ? If it is possible to transfer the mission data before, it would save battery.

Reply:
That's a good question.
This is how it is done in the example project but we could change it.